build.sbt

```diff
@@ -90,12 +14,22 @@ val newerCompilerLintSwitches = Seq(
   "-Ywarn-numeric-widen" // In 2.10 this produces a some strange spurious error
 )
 
-scalacOptions ++= allVersionCompilerLintSwitches
-
-scalacOptions in (Compile, console) ~= (_ filterNot (Set("-Xfatal-warnings", "-Ywarn-unused-import").contains))
-
-scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value
-
-scalacOptions ++= PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)){
+lazy val settings = Seq(
+  scalaVersion := "2.12.0",
+  crossScalaVersions := Seq("2.10.5", "2.11.8", "2.12.0"),
+  scalacOptions ++= Seq("-feature"),
+  scalacOptions ++= allVersionCompilerLintSwitches,
+  scalacOptions in (Compile, console) ~= (_ filterNot (Set("-Xfatal-warnings", "-Ywarn-unused-import").contains)),
+  scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value,
+  scalacOptions ++= PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)){
     case Some((2, scalaMajor)) if scalaMajor >= 11 => newerCompilerLintSwitches
-}.toList.flatten
+  }.toList.flatten,
+  initialize := {
+    val required = "1.8"
```

Comment:
The target `jvm-1.8` [has been added from Scala `2.11.5`](https://issues.scala-lang.org/browse/SI-8966), which means that we cannot use it because we still compile for Scala `2.10.5`